File: test_archives/py_tests/batch4/test_function7.py
from functions.function7 import function7
import pytest 

def test_function7_1(): 
    assert function7(1, "abc", 2.5, ["a", "b"]) == 4.5

def test_function7_2(): 
    assert function7(3, "xyz", 3.5, ["x", "y"]) == 12.5

File: test_archives/cpp_tests/batch4/test_function7.cpp
#include "../catch.hpp"
#include "../functions/function7.cpp"

TEST_CASE( "Testing function7", "[function7]" ) {

    SECTION( "Test function7_1" ) {
        REQUIRE( function7(1, "abc", 2.5, {"a", "b"}) == 4.5 );
    }

    SECTION( "Test function7_2" ) {
        REQUIRE( function7(3, "xyz", 3.5, {"x", "y"}) == 12.5 );
    }

}